task-sdk/src/airflow/sdk/definitions/asset/__init__.py

```diff
@@ -166,15 +166,16 @@ def _sanitize_uri(inp: str | ObjectStoragePath) -> str:
         return uri
     if normalized_scheme == "airflow":
         raise ValueError("Asset scheme 'airflow' is reserved")
-    _, auth_exists, normalized_netloc = parsed.netloc.rpartition("@")
-    if auth_exists:
+    if parsed.password:
         # TODO: Collect this into a DagWarning.
         warnings.warn(
-            "An Asset URI should not contain auth info (e.g. username or "
-            "password). It has been automatically dropped.",
+            "An Asset URI should not contain a password. User info has been automatically dropped.",
             UserWarning,
             stacklevel=3,
         )
+        normalized_netloc = parsed.netloc.split("@", 1)[1]
```

Comment:
```suggestion
        _, _, normalized_netloc = parsed.netloc.rpartition("@")
```

According to the spec, `@` must be split from the right, not left.